src/main/java/gov/nasa/jpl/mbee/mdk/mms/actions/CommitDiagramArtifactsAction.java

```diff
@@ -54,37 +56,34 @@ public void execute(Collection<Annotation> annotations) {
 
     @Override
     public void actionPerformed(@CheckForNull ActionEvent actionEvent) {
-        URIBuilder artifactsUriBuilder = MMSUtils.getServiceProjectsRefsArtifactsUri(project);
-        if (artifactsUriBuilder == null) {
-            return;
-        }
-        URI artifactsUri;
-        try {
-            artifactsUri = artifactsUriBuilder.build();
-        } catch (URISyntaxException e) {
-            Application.getInstance().getGUILog().log("[ERROR] Unable to commit artifact " + Converters.getElementToHumanNameConverter().apply(diagram) + ". Reason: " + e.getMessage());
-            e.printStackTrace();
+        MMSEndpoint mmsArtifactsEndpoint = MMSUtils.getServiceProjectsRefsArtifactsUri(project);
+        if (mmsArtifactsEndpoint == null) {
             return;
         }
         TaskRunner.runWithProgressStatus(progressStatus -> {
             try {
-                URIBuilder elementsRequestUri = MMSUtils.getServiceProjectsRefsElementsUri(project);
-                if (elementsRequestUri == null) {
+                MMSEndpoint mmsElementsEndpoint = MMSUtils.getServiceProjectsRefsElementsUri(project);
+                if (mmsElementsEndpoint == null) {
                     return;
                 }
                 for (MMSArtifact artifact : artifacts) {
-                    HttpPost request = new HttpPost(artifactsUri);
                     HttpEntity entity = MultipartEntityBuilder.create().addTextBody(MDKConstants.ID_KEY, artifact.getId()).addTextBody(MDKConstants.CHECKSUM_KEY, artifact.getChecksum()).addTextBody("source", "magicdraw").addBinaryBody("file", artifact.getInputStream(), artifact.getContentType(), artifact.getId() + ".tmp").build();
-                    request.setEntity(entity);
-                    MMSUtils.sendMMSRequest(project, request, progressStatus);
+                    File file = File.createTempFile(this.getClass().getSimpleName() + "-" + ContentType.APPLICATION_JSON.getMimeType().replace('/', '-') + "-", null);
```

Comment:
Guessing this is borrowed from `MMSUtils.createEntityFile` and that is why `ContentType.APPLICATION_JSON` shows up, but I don't think it applies in the artifact context. I think there's also an opportunity to split `createEntityFile` into a few methods and handle this there.

Not a bad idea, but I'm curious why we're dumping the artifact to file and then streaming it in a request. This is typically a good strategy to handle arbitrarily large requests, but in this case the Cameo API generates the diagram to memory so it's already taking up the space.